lib/lists.nix

```
@@ -687,6 +687,47 @@ rec {
    */
   unique = foldl' (acc: e: if elem e acc then acc else acc ++ [ e ]) [];
 
+  /* Check if list contains only unique elements.
+
+     Type: allUnique :: [a] -> bool
+
+     Example:
+       allUnique [ 3 2 3 4 ]
+       => false
+       allUnique [ 3 2 4 1 ]
+       => true
+   */
+  allUnique = lst: (builtins.length (unique lst) == builtins.length lst);
+
+  /* Check unique elements by sorting the list with a comparator and checking uniqueness with the previous element.
+     O(n) complexity on top of `sort`.
```

```suggestion
  /* Sort a list using the `<` operator and remove any duplicate elements.
     O(n log n) complexity.
```

I think it's better to document the overall complexity.

---

done